infra/modules/azure_function_app/variables.tf

```diff
@@ -0,0 +1,148 @@
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "env_short" {
+  type        = string
+  description = "Environment short name"
+}
+
+variable "location" {
+  type        = string
+  description = "Location name"
+  default     = "italynorth"
+}
+
+variable "prefix" {
+  type        = string
+  description = "Project prefix"
+}
+
+variable "instance_number" {
+  type        = string
+  description = "(Optional) Instance count for specific resources"
+  default     = "01"
+
+  validation {
+    condition     = can(regex("^(0[1-9]|[1-9][0-9])$", var.instance_number))
+    error_message = "The variable \"instance_number\" only accepts values in the range [01-99] as strings"
+  }
+}
+
+variable "domain" {
+  type        = string
+  description = "Domain of the project"
+}
+
+variable "app_name" {
+  type        = string
+  description = "Name of this single application"
+
+  validation {
+    condition     = length(var.app_name) > 1
+    error_message = "The variable \"app_name\" must contain at least 2 characters"
+  }
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+variable "app_service_plan_id" {
+  type        = string
+  default     = null
+  description = "(Optional) Set the AppService Id where you want to host the Function App"
+}
+
+variable "application_insights_connection_string" {
```

Comment:
should be optional in case you want to use only the sdk integration

Reply:
Updated